internal/model/function_test.go

```diff
@@ -0,0 +1,243 @@
+package model_test
+
+import (
+	"testing"
+
+	. "github.com/onsi/gomega"
+
+	"github.com/go-logr/logr"
+
+	"github.com/theunrepentantgeek/crddoc/internal/config"
+	"github.com/theunrepentantgeek/crddoc/internal/model"
+	"github.com/theunrepentantgeek/crddoc/internal/packageloader"
+)
+
+// assertHasParameter asserts that a parameter list contains a parameter with the given name and type.
+func assertHasParameter(t *testing.T, g *GomegaWithT, params []model.Parameter, name, typeName string) {
+	t.Helper()
+	g.Expect(params).NotTo(BeEmpty(), "parameter list should not be empty")
+	
+	found := false
+	for _, p := range params {
+		if p.Name == name && p.Type.Name() == typeName {
+			found = true
+			break
+		}
+	}
+	g.Expect(found).To(BeTrue(), "expected to find parameter %s of type %s", name, typeName)
+}
+
+// assertHasResult asserts that a result list contains a result with the given type.
+func assertHasResult(t *testing.T, g *GomegaWithT, results []model.Parameter, typeName string) {
```

Comment:
Prefer to put helper methods like this at the end of the file, so that readers see the tests first. They can infer the meaning of the helper method from the name.

Add a `resultName` property just prior `typeName` so that we can do asserts for named return types too. Update usage where needed.

Also, I'm reversing my prevous request - remove the `g` parameter and create your own gomega helper within both this method and `assertHasParameter`.